### PR TITLE
Speedup oscilloscope

### LINKED ISSUE #148: Add oscillosope speedup mode
- PopupMenu in Settings with resampling methods: Normal, Mean, Min/Max, Min/Max (rowloc)
- Add to config code and config template
- Only for DAQ for all hardware types

### PR DESCRIPTION
**Pull request by t-b**
Close #148.

First pre-release.

The speedups can be seen best from the following graph:

Graph0

- [x] Proper commits
- [x] Better documentation for decimation methods
- [x] NI support
- [x] move controls to osci group, create sub-group boxes
- [x] Add config file entries
- [x] Add support for automatic decimation factor calculation from oscilloscope width using decFactor == -1
- [x] disallow dec factor 0 and 1 from DAP_CheckSettings.
- [x] Finalize "TP during DAQ" changes
- [x] Introduce a new wave GetDataWave() which holds the scaled data at all times. This can replace the oscilloscope data wave in the analysis functions. Add a new member to `AnalysisFunctions_V3` structure which points to that wave and use that in the analysis functions. Lock it as well.
- [x] Commit cleanup. 
 - [x] Introduce scaledDatawave earlier than decimation
 - [x] Add documentation for analysis function scaledDataWave, check in tests
 - [x] Add some more checks for the sweep wave in `AllTests`

### CODE REVIEW

File: Packages/MIES/MIES_MiesUtilities.ipf [hunk 1]
```diff
@@ -5447,3 +5447,182 @@ Function StopAsyncIfDone()
...
+	if(first > lastRow)
+		return NaN
+	endif
+
+	last  = min(last, lastRow)
+
+	targetFirst = row * 2
+	targetLast = (row * 2) + 1
+
+	WaveStats/Q/M=1/RMD=[first, last][col] input
```
[the remark this answers is not in the document]
Comment (t-b) [reply Packages/MIES/MIES_MiesUtilities.ipf]: @timjarsky Yes, according to tests Adam sent me and I verified. IP9 will have a specialized matrixOP function for it. I can do some tests with IgorCL (the calculate on graphics card XOP) if it is still too slow.

**Comment by t-b** [comment]
@timjarsky Ready for testing. I've added all points from our chat.

**Comment by timjarsky** [comment]
> @timjarsky Ready for testing. I've added all points from our chat.

@t-b I've tested and have passed it along to Stephanie and Lisa for testing (I'm expecting to uncover differences between our test rigs and production rigs),

**Comment by t-b** [comment]
@timjarsky Ready for testing.

**Comment by t-b** [comment]
@timjarsky This also has the change that I reuse the scaled data wave at the end of the sweep for the real sweep data. Here this makes finishing the sweep smoother.

**Comment by timjarsky** [comment]
@t-b I've run this PR with 8 channels on my test set-up and it looks fine but I'll wait for the tests to pass before testing on a production rig,

**Comment by timjarsky** [comment]
@t-b

**Comment by t-b** [comment]
@timjarsky Stupid mistake, should be fixed.

**Comment by timjarsky** [comment]
@t-b the default decimation looks a little aggressive:

**Comment by t-b** [comment]
@timjarsky THanks for the nice wording. Yes instead of taking twice the pixels I took half the pixels. Should be fixed now.

**Comment by t-b** [comment]
@timjarsky The failing tests were failing for real. I wanted to sneak in some additional tests, but I have left them out for now.

**Comment by t-b** [comment]
@timjarsky Ready to go. The ITC1600 issues are still on investigation but not related to this PR.

**Pull request merged**